### ISSUE TITLE
Intellisense regression for lambda parameters constrained to struct

### ISSUE THREAD

**Issue by jbevain** [dotnet/roslyn#43279]
**Version Used**: 
Verified in VS 16.5, VSM 8.5
Regression from 16.4/8.4

**Steps to Reproduce**:
Supporting code:

```csharp
public interface IComponentData
{
}
public class EntitiesQueryBuilder
{
	public delegate void F_D<T0>(ref T0 d0) where T0 : struct, IComponentData;
	public void ForEach<T0>(F_D<T0> f) where T0 : struct, IComponentData
	{
	}
}
public abstract class Component
{
	public EntitiesQueryBuilder Entities => null;
	protected abstract void OnUpdate();
}
public struct Person : IComponentData
{
	public float HP;
}
public class PersonComponent : Component
{
	protected override void OnUpdate()
	{
		Entities.ForEach((ref Person p) =>
		{
		});
	}
}
```

1. Paste code in a C# project
2. Go to the `OnUpdate` method in `PersonComponent`
3. Go into the lambda passed to the ForEach call.
4. Type `p.` and ask for completion.

**Expected Behavior**:
HP field should be available in the IntelliSense.

**Actual Behavior**:
HP field is not visible anymore in the list presented by IntelliSense.

This is a best effort repro from the Unity APIs. This has been originally reported on the [Unity forums](https://forum.unity.com/threads/ecs-intellisense-malfunctioning-work-around.861274/)

See picture:

![image](https://user-images.githubusercontent.com/79284/79031960-9fb48300-7b57-11ea-97a2-c57aa28cdc0b.png)

This might be related to  #42910.

**Comment by jbevain** [comment]
Cut down repro:

```
public struct Person
{
	public float HP;
}

public class PersonComponent
{
	protected void OnUpdate()
	{
		ForEach((Person p) =>
		{
		});
	}

	public delegate void F_D<T0>(T0 d0) where T0 : struct;

	public void ForEach<T0>(F_D<T0> f) where T0 : struct
	{
	}
}
```

**Comment by jbevain** [comment]
Good news, that seems to be fixed in 16.6.

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/42942
